### PR TITLE
parser: add new ddl action AlterIndexVisibility

### PR DESCRIPTION
**Pull request by Deardrops**
### What problem does this PR solve? 

For support `alter table ... alter index ... invisible` statement, we add a new DDL action `AlterIndexVisibility`.

### What is changed and how it works?

- [x] Add a new DDL action `Enum`
- [x] Add a new field `KeyName` (`model.CIStr`) in struct `AlterTableSpec`, it will replace the `Name` (`string`) field in future PR.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

 - None

Related changes

 - None

### CODE REVIEW

**Comment by wjhuang2016** [comment]
Need to cherry-pick to the release branch？

**Comment by wjhuang2016** [comment]
Any update for parser.y?

**Comment by reafans** [comment]
LGTM

**Comment by Deardrops** [comment]
> 
> 
> No test cases?

@djshow832 Test has covered in #519

**Comment by Deardrops** [comment]
> 
> 
> Need to cherry-pick to the release branch？

@wjhuang2016 yes, but I don't find label `cherry-pick 4.0`.

**Comment by Deardrops** [comment]
> 
> 
> Any update for parser.y?

@wjhuang2016 see https://github.com/pingcap/parser/pull/840/files#diff-f615844d3497ff38db57e459d6ef657bR10329

**Comment by Deardrops** [comment]
> 
> 
> Please remove `Need to cherry-pick to the release branch`, rest LGTM

@wjhuang2016 Done

File: ast/ddl.go [hunk 1]
```diff
@@ -2245,6 +2245,7 @@ type AlterTableSpec struct {
 
 	Tp              AlterTableType
 	Name            string
+	KeyName         model.CIStr
```
Comment (wjhuang2016) [inline ast/ddl.go:2248]: How about "IndexName"?

Comment (Deardrops) [reply ast/ddl.go]: @wjhuang2016 Done, PTAL

**Pull request merged**